plugin/Plugin_TMDojo.as

```diff
@@ -566,21 +579,86 @@ void RenderMenu()
             }
         }
 
+        if (pluginAuthed) {
+            if (UI::MenuItem(green + Icons::Plug + " Plugin Authenticated")) {
+            }
+        } else {
+            if (UI::MenuItem(red + Icons::Plug + " Authenticate Plugin")) {
+                authenticatePlugin();
+            }
+        }
+
 		UI::EndMenu();
 	}
 }
 
+void getPluginAuth() {
+    while (checkSessionIdCount < maxCheckSessionId) {
+        sleep(1000);
+        print("getPluginAuth Count: " + checkSessionIdCount + " / " + maxCheckSessionId);
+        checkSessionIdCount++;
+        Net::HttpRequest@ auth = Net::HttpGet(ApiUrl + "/auth/pluginSecret?clientCode=" + ClientCode);
+        while (!auth.Finished()) {
+            yield();
+            sleep(50);
+        }
+        print("getPluginAuth response: " + auth.String());
+        try {
+            Json::Value json = Json::Parse(auth.String());
+            SessionId = json["sessionId"];
+            UI::ShowNotification("TMDojo", "Plugin is authenticated!", vec4(0, 0.4, 0, 1));
+            pluginAuthed = true;
+            ClientCode = "";
+            break;
+        } catch {
+            error("/auth/pluginSecret: no sessionId in reponse");
+        }
+    }
+    if (checkSessionIdCount >= maxCheckSessionId) {
+        UI::ShowNotification("TMDojo", "Plugin authentication took too long, please try again", vec4(0.4, 0, 0, 1));
+    }
+}
+
+void authenticatePlugin() {
+    OpenBrowserURL(pluginAuthUrl);
+    startnew(getPluginAuth);
+}
+
 void checkServer() {
     g_dojo.checkingServer = true;
     g_dojo.playerName = g_dojo.network.PlayerInfo.Name;
     g_dojo.playerLogin = g_dojo.network.PlayerInfo.Login;
     g_dojo.webId = g_dojo.network.PlayerInfo.WebServicesUserId;
-    Net::HttpRequest@ auth = Net::HttpGet(ApiUrl + "/auth?name=" + g_dojo.playerName + "&login=" + g_dojo.playerLogin + "&webid=" + g_dojo.webId);
+    Net::HttpRequest@ auth = Net::HttpGet(ApiUrl + "/auth?name=" + g_dojo.playerName + "&login=" + g_dojo.playerLogin + "&webid=" + g_dojo.webId + "&sessionId=" + SessionId);
```

Comment:
In the future, we should also include the `sessionId` for the other requests, but I believe they are not used yet so it's no problem now.